packages/dds/tree/docs/user-facing/array-merge-semantics.md

```diff
@@ -0,0 +1,333 @@
+For example, if one user moves item B leftward to the start of array `[A, B, C]`,
+and some other user concurrently moves item B rightward to the end of the array,
+then the item will be affected by each successive move in sequencing order:
+* If the leftward move is sequenced before the rightward move
+  then A will first be moved to the start (thus yielding `[B, A, C]`)
+  then moved to the end (thus yielding `[A, C, B]`).
+* If the rightward move is sequenced before the leftward move
+  then A will first be moved to the end (thus yielding `[A, C, B]`)
+  then moved to the start (thus yielding `[B, A, C]`).
+
+A removed item may be restored as a result of a concurrent move operation
+and a moved item may be removed as a result of a concurrent remove operation.
+For example, if one user moves item A to the end of array `[A, B, C]`,
+and some other user concurrently removes item A,
+then the item will be affected by each successive operation in sequencing order:
+* If the move is sequenced before the remove
+  then A will first be moved to the end of the array (thus yielding `[C, B, A]`)
+  then removed (thus yielding `[C, B]`).
+* If the remove is sequenced before the move
+  then A will first be removed (thus yielding `[C, B]`)
+  then moved (thus yielding `[C, B, A]`).
+
+## Core Editing Operations
+
+### `insertAt(gapIndex: number, ...value: readonly (TNew | IterableTreeArrayContent<TNew>)[]): void`
+
+Inserts new items at the location described by `gapIndex`.
+
+Preconditions:
+* There is no schema change edit that this edit is both concurrent to and sequenced after.
+* The inserted values must have status `TreeStatus.New` or be primitives.
+  (This precondition will be removed soon)
+
+Postconditions:
+* The values are inserted in the targeted [gap](#location-of-inserted-items).
+
+### `moveRangeToIndex(destinationGap: number, sourceStart: number, sourceEnd: number, source: TMoveFrom): void`
+
+Moves the specified items from the given source array to the desired location within the array.
+
+Preconditions:
+* There is no schema change edit that this edit is both concurrent to and sequenced after.
+
+Postconditions:
+* The [specified items](#specifying-the-set-of-removed-items) are moved to the targeted [gap](#location-of-inserted-items).
+
+If multiple clients concurrently move an item,
+then that item will be moved to the destination indicated by the move of the client whose edit is ordered last.
+
+### `removeRange(start?: number, end?: number): void`
+
+Removes the items between the specified indices.
+
+Preconditions:
+* There is no schema change edit that this edit is both concurrent to and sequenced after.
+
+Postconditions:
+* The [specified items](#specifying-the-set-of-removed-items) are removed.
+
+Removed items are saved internally for a time in case they need to be restored as a result of an undo operation.
+Changes made to them will apply despite their removed status.
```

Comment:
It's a bit more complicated than that because undo operation can also edit a removed item (even when they don't restore that item), but overall I think your suggestion is the safer one.